packages/alchemy/src/react/hooks/useAccount.ts

```diff
@@ -0,0 +1,59 @@
+"use client";
+
+import { useMutation } from "@tanstack/react-query";
+import { useEffect, useSyncExternalStore } from "react";
+import type { SupportedAccount, SupportedAccountTypes } from "../../config";
+import { createAccount } from "../../config/actions/createAccount.js";
+import {
+  getAccount,
+  type GetAccountParams,
+} from "../../config/actions/getAccount.js";
+import { watchAccount } from "../../config/actions/watchAccount.js";
+import { AlchemySignerStatus } from "../../signer/index.js";
+import { useAlchemyAccountContext } from "../context.js";
+import { useSignerStatus } from "./useSignerStatus.js";
+
+export type UseAccountResult<TAccount extends SupportedAccountTypes> = {
+  account?: SupportedAccount<TAccount>;
+  isLoadingAccount: boolean;
+};
+
+export function useAccount<TAccount extends SupportedAccountTypes>(
+  params: GetAccountParams<TAccount> & { skipCreate?: boolean }
+): UseAccountResult<TAccount> {
+  const { config, queryClient } = useAlchemyAccountContext();
+  const status = useSignerStatus();
+  const account = useSyncExternalStore(
+    watchAccount(params.type, config),
+    () => getAccount(params, config),
+    () => undefined
+  );
+
+  const { mutate, isPending } = useMutation(
+    {
+      mutationFn: async () => account?.account ?? createAccount(params, config),
+      mutationKey: ["createAccount", params.type],
+    },
+    queryClient
+  );
+
+  useEffect(() => {
+    if (
+      !params.skipCreate &&
+      status === AlchemySignerStatus.CONNECTED &&
+      !account &&
+      !isPending
+    ) {
+      mutate();
+    }
+  }, [account, isPending, mutate, params.skipCreate, status]);
```

Comment:
I'm gonna try condensing this into a `useQuery` call instead of this.

The reason I had to do this instead of the `getAccount` method was because if I set state in there, then I'd end up calling setState during the render loop. This was causing an error. Having it in a `useEffect` guarantees that `setState` is not called during the render loop